docs/blog/pydantic-v2.md

```diff
@@ -319,7 +319,42 @@ translate it to Rust.
 We should also add support for `validate_alias` and `dump_alias` as well as the standard `alias`
 to allow for customising field keys.
 
-### Model namespace cleanup :thumbsup:
+### Validation Context :thumbsup:
+
+Pydantic V2 will add a new optional `context` argument to `model_validate` and `model_validate_json`
+which will allow you to pass information not available when creating a model to validators.
+See [pydantic#1549](https://github.com/samuelcolvin/pydantic/issues/1549) for motivation.
+
+```py title="Context during Validation"
+from pydantic import BaseModel, EmailStr, validator
+
+class User(BaseModel):
+    email: EmailStr
+    home_country: str
+    
+    @validator('home_country')
+    def check_home_country(cls, v, context):
+        if v not in context['countries']:
+            raise ValueError('invalid country choice')
+        return v
+
+async def add_user(post_data: bytes):
+    countries = set(await db_connection.fetch_all('select code from country'))
+    user = User.model_validate_json(post_data, context={'countries': countries})
+    ...
+```
+
+!!! note
+    We (actually mostly Sebastián :wink:) will have to make some changes to FastAPI to fully leverage `context`
+    as we'd need some kind of dependency injection to build context before validation so models can still be passed as
+    arguments to views. I'm sure he'll be game.
```

Comment:
I'm game! Haha :sweat_smile: 

I think it's gonna be challenging but I'm confident is solvable/doable.

Reply:
we might need to add another method to models like

```py
class MyModel(BaseModel):
    @classmethod
    async def model_prepare_context(cls, db_conn: Depends(get_db)) -> dict[str, Any:
        """allows custom logic to prepare context"""
```

I guess `model_prepare_context` doesn't even need to on `BaseModel`, you just document in FastAPI that if it's defined, it'll be called with dependency resolution to prepare context?

Up to you, but I'm very happy to add something like this to pydantic if it helps FastAPI.